### PR TITLE
MAISTRA-2606 Add missing commits from 2.0

### PR DESCRIPTION
**Pull request by luksa**

### CODE REVIEW

File: pkg/controller/servicemesh/controlplane/profiles_test.go [hunk 1]
```diff
@@ -160,6 +160,37 @@ func TestProfiles(t *testing.T) {
 				smcp.Spec.TechPreview.RemoveField("wasmExtensions")
 			},
 		},
+		{
+			name: "MAISTRA-2409",
+			v:    versions.V2_0,
```
Comment (jwendell) [inline pkg/controller/servicemesh/controlplane/profiles_test.go:165]: do we need to duplicate this test with a 2.1 version?

Comment (luksa) [reply pkg/controller/servicemesh/controlplane/profiles_test.go:165]: Not really, as the code is the same for both 2.0 and 2.1. If anything, we'd have to extend all the tests to check for both 2.0 and 2.1, but that's a whole new PR.

Comment (dgn) [reply pkg/controller/servicemesh/controlplane/profiles_test.go:165]: I recently filed an issue to cover that: https://issues.redhat.com/browse/MAISTRA-2574

**Review by dgn — APPROVED** [review]

File: pkg/controller/servicemesh/controlplane/profiles_test.go (hunk 1 above)
Comment (jwendell) [reply pkg/controller/servicemesh/controlplane/profiles_test.go:165]: The code is the same, but the charts not.

**Review by jwendell — APPROVED** [review]

**Pull request merged**